eclipse-collections/src/main/java/org/eclipse/collections/impl/collector/Collectors2.java

```diff
@@ -878,6 +880,29 @@ private Collectors2()
                 EMPTY_CHARACTERISTICS);
     }
 
+    /**
+     * Groups the elements using the {@code groupBy} function and all the elements that map to the same key are
+     * aggregated together using the {@code aggregator} function. The third parameter, the {@code zeroValueFactory}
+     * function, creates the initial value in each aggregation. Aggregate results are allowed to be immutable as they
+     * will be replaced in the map.
+     */
+    public static <T, K, R extends MutableMap<K, T>> Collector<T, ?, R> aggregateBy(
+            Function<? super T, ? extends K> groupBy,
+            Function0<? extends T> zeroValueFactory,
+            Function2<? super T, ? super T, ? extends T> aggregator,
+            Supplier<R> supplier)
+    {
+        return Collector.of(
+                supplier,
+                (map, each) -> new NonMutatingAggregationProcedure<T, K, T>(map, groupBy, zeroValueFactory, aggregator).value(each),
```

Comment:
Calling new on NonMutatingAggregationProcedure will create linear garbage.  Inline the code here from that class.

Reply:
Please see if you can inline the code here so we don't have to create a new NonMutatingAggregationProcedure on every call to the accumulator.